src/XMakeBuildEngine/UnitTests/Evaluation/ImportFromMSBuildExtensionsPath_Tests.cs

```diff
@@ -167,6 +168,47 @@ public void ImportFromExtensionsPathWithWildCard()
         }
 
         [Fact]
+        public void ImportFromExtensionsPathWithWildCardAndSelfImport()
```

Comment:
You need to add a 3rd proj file which gets imported after the `main.proj` attempt. So, maybe from a `extnDir3`. That way we would test the new behavior of the search+import not stopping when a circular import is hit.